### DISCUSSION TITLE
How to change font size and line spacing of DetailedList on android app?

### DISCUSSION THREAD

**Discussion by ccinsz** [beeware/briefcase#1600] [Q&A]
The font size and line spacing of DetailedList on android app is so big, how to change it?
![4](https://github.com/beeware/briefcase/assets/134466342/5e938e9f-356e-4ef9-92d9-58b28c4f2976)

**Comment by freakboy3742** [comment]
This is really a Toga question, rather than a Briefcase question - but t\The font and line spacing of DetailedList isn't currently configurable. The excessive whitespace you're seeing is mostly due to the fact that the icons aren't rendering. 

![Screenshot_1705041106](https://github.com/beeware/briefcase/assets/37345/04dc0280-5286-4099-a281-e54b48ca5853)

That said: even allowing for the icons, the padding in the Android detailed list is a little *too* generous; I've opened beeware/toga#2338 to improve the default padding.

**Reply by ccinsz** [reply]
The Selection also has the problem of line spacing，please improve it, too.
![5](https://github.com/beeware/briefcase/assets/134466342/2f8b5972-e1ef-427e-9db7-3ef3871628a0)

**Reply by freakboy3742** [reply]
That's the default spacing of an Android Spinner widget. Unlike the DetailedList, that's not something where we're configuring the sizing. 

If I had to guess, it's that size to ensure that the target for a finger is large enough to be easy to hit.

**Reply by ccinsz** [reply]
Got it. Please tell me if the spacing of the DetailedList is solved.

**Reply by freakboy3742** [reply]
You can follow the progress of the DetailedList change on the PR I linked previously.

**Reply by ccinsz** [reply]
Ok, thank you for your help.

**Reply by ccinsz** [reply]
Does the DetailedList not set the border or the size of font? Now the size of font is a little big and the DetailedList does not the border which make the user does not the height of it.

**Reply by freakboy3742** [reply]
There are no border style controls in Toga, so I don't know what you're referring to there.

However, you're correct that DetailedList doesn't currently honor font style attributes. This is at least in part because there are 2 fonts that need to be defined, and we don't currently have a way to target *parts* of a widget for styling.

As for the default font being "a little big"... I don't know if you're seeing something different, but in my testing, the size seems entirely appropriate:

<img width="431" alt="Screenshot 2024-01-15 at 12 15 13 pm" src="https://github.com/beeware/briefcase/assets/37345/e15835cd-7952-4bc6-b8bf-aa0618993f97">

The screenshot you gave previously seems to confirm you're seeing the same thing. If you're not, provide details and we can look into the problem.

**Reply by ccinsz** [reply]
If there are more than 2 lines in the subtitle of DetailedList , the display will be incomplete. And the icon is set None，there will be a blank space on the left side, which looks very unfriendly.
![7](https://github.com/beeware/briefcase/assets/134466342/3892b1c2-c51c-464f-b8b4-4e4f6f1f71e0)

**Reply by freakboy3742** [reply]
That's not a font size problem. There's no font size that would allow for arbitrary number of lines of text to fit, *and* guarantee the text is legible.

This is a layout problem - and in this case, an indication of a bug in the Android implementation. At present, the DetailedList API only guarantees a single line of text will be rendered for both the title and subtitle. Android apparently attempts to render multiple lines of text; that's not the intended behavior. It *should* be clipping the title and subtitle to a single line. I've opened beeware/toga#2343 to track this issue.

The icon is currently non-optional in the layout. I'd be open to an API change that makes the icon optional in a layout. I've opened beeware/toga#2342 to track this feature proposal.
